app/models/dialog.rb

```diff
@@ -156,6 +162,10 @@ def dialog_field_hash
     @dialog_field_hash ||= dialog_fields.each_with_object({}) { |df, hash| hash[df.name] = df }
   end
 
+  def check_for_duplicate_dialog_fields(dialog_field_list)
+    dialog_field_list.empty? ? [] : dialog_field_list.group_by(&:name).select { |_k, v| v.size > 1 }.map(&:first).join(",")
```

Comment:
You may want to look at https://github.com/ManageIQ/more_core_extensions/#array `#element_counts`

Reply:
That's magical, thanks.